lib/cartopy/mpl/feature_artist.py

```diff
@@ -179,15 +180,22 @@ def draw(self, renderer, *args, **kwargs):
                 geom_key, geom)
             mapping = FeatureArtist._geom_key_to_path_cache.setdefault(
                 geom_key, {})
-            geom_paths = mapping.get(key)
-            if geom_paths is None:
+            geom_path = mapping.get(key)
+            if geom_path is None:
                 if ax.projection != feature_crs:
                     projected_geom = ax.projection.project_geometry(
                         geom, feature_crs)
                 else:
                     projected_geom = geom
+
                 geom_paths = cpatch.geos_to_path(projected_geom)
-                mapping[key] = geom_paths
+                if not geom_paths:
+                    continue
+                elif len(geom_paths) > 1:
+                    geom_path = mpath.Path.make_compound_path(*geom_paths)
+                else:
+                    geom_path, = geom_paths
```

Comment:
do you need the special case, or can you input a single path into it as well?
```suggestion
                # The transform may have split the geometry into two paths, we only want one compound path
                geom_path = mpath.Path.make_compound_path(*geom_paths)
```

Should this logic be moved up into the `geos_to_path()` function though? That currently returns a list of paths, but it seems like if you're passing in one geometry that should be a compound path like you have here.

Reply:
It tried pushing this to `geos_to_path`, but got failures because sometimes `projected_geom` is a tuple, having been transformed from a `LinearRing`
https://github.com/SciTools/cartopy/blob/25a2154604517cc6d46bbb82b83f1c46cb0df381/lib/cartopy/crs.py#L907

Reply:
sigh :( Thanks for looking into that! I think the geometry transforms needs a good rework with all the great fixes you're coming up with! Looking through the linearring code, it looks like it is primarily used within `_project_polygon` to project the boundaries and holes and then attaches the dangling linestring to the boundary. I wonder if we could just return a single MultiLineString in that case and consider the start/end ourselves for whether it is closed (LinearRing) or not.

This is good as-is, I'm just writing down thoughts for future ideas at this point.